Public/Src/Engine/Dll/Distribution/RemoteWorker.cs

```diff
@@ -759,27 +761,28 @@ private ExecutionResult LogAndGetNetworkFailureResult(PipCompletionTask pipCompl
                     runnablePip.Description,
                     Name,
                     errorMessage: errorMessage,
-                    step: runnablePip.Step.AsString());
+                    step: runnablePip.Step.AsString(),
+                    callerName: callerName);
 
                 // Return success result
                 result = new ExecutionResult();
                 result.SetResult(operationContext, PipResultStatus.NotMaterialized);
                 result.Seal();
-                pipCompletionTask.Set(result);
+
+                pipCompletionTask.TrySet(result);
+                return;
             }
-            else
-            {

```

Comment:
So I guess the removal of this else is what causes an error to be logged and fixes the crash. Am I following correctly?

Reply:
An error is logged and the pip is marked as NotMaterialized, so not a failure. What ends up happening downstream?

1. Will that still keep our pip result <-> error logged validation out of agreement?
1. How will the consumers of that pip's output behave?